### PR TITLE
[SPARK-30478][CORE][DOCS] Fix Memory Package documentation

### PR DESCRIPTION
**Pull request by sddyljsx**
### What changes were proposed in this pull request?
update the doc of momery package

### Why are the changes needed?
From Spark 2.0, the storage memory also uses off heap memory. We update the doc here.
memory manager

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
No Tests Needed

### CODE REVIEW

**Comment by sddyljsx** [comment]
@srowen PTAL

File: core/src/main/scala/org/apache/spark/memory/package.scala [hunk 1]
```diff
@@ -40,25 +40,26 @@ package org.apache.spark
...
+ *                                                        +---->|                           |
+ *                                        *               |     |  +---------------------+  |
+ *                                        *               |     |  |OnHeapExecMemPool    |  |
+ *       +-------------+                  *               |     |  +---------------------+  |
+ *       | MemConsumer |----+                             |     |                           |
+ *       +-------------+    |    +-------------------+    |     |  +---------------------+  |
+ *                          +--->| TaskMemoryManager |----+     |  |OffHeapExecMemPool   |  |
+ *                               +-------------------+          |  +---------------------+  |
+ *                                                              |                           |
+ * }}}                                                          +---------------------------+
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/memory/package.scala]: I think the open and close braces have to be on their own lines. Here, the last part of the table is outside the block.

**Review by srowen — CHANGES_REQUESTED** [review]

File: core/src/main/scala/org/apache/spark/memory/package.scala (hunk 1 above)
Comment (sddyljsx) [reply core/src/main/scala/org/apache/spark/memory/package.scala]: Fixed

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Jenkins test this please

**Comment by srowen** [comment]
Jenkins retest this please

**Comment by srowen** [comment]
I think that's a known issue with Python tests, not related. Let me wait a bit.

**Comment by sddyljsx** [comment]
> Let me wait a bit

ok, thanks for review.

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @sddyljsx and @srowen .
I merge this PR because this is a documentation PR.
Merged to master/2.4.

**Comment by sddyljsx** [comment]
Thanks! @srowen @dongjoon-hyun

**Pull request closed without merging**